packages/sku/src/services/vite/preload/collector.ts

```diff
@@ -0,0 +1,245 @@
+      preloads: this.preloadIds,
+      scripts: this.scriptIds,
+      nonce,
+    });
+  }
+
+  register(moduleId: ModuleId) {
+    this.moduleIds.add(moduleId);
+    parseManifestForEntry({
+      manifest: this.manifest,
+      entry: moduleId,
+      preloads: this.preloadIds,
+      scripts: this.scriptIds,
+      nonce: this.nonce,
+    });
+  }
+  getAllModules() {
+    return [...this.moduleIds];
+  }
+  getAllPreloads() {
+    const preloadHtml = [...this.preloadIds.values()]
+      .sort(sortPreloads)
+      .map(createHtmlTag)
+      .join('\n');
+
+    return preloadHtml;
+  }
+  getAllScripts() {
+    const scriptHtml = [...this.scriptIds.values()]
+      .sort(sortInjectableScript)
+      .map(createScriptTag)
+      .join('\n');
+
+    return scriptHtml;
+  }
+  getAllLinks() {
+    const linkTags = [...this.preloadIds.values()]
+      .sort(sortPreloads)
+      .map(createLinkTag)
+      .filter(Boolean);
+
+    return linkTags;
+  }
+}
+
+const parseManifestForEntry = ({
+  manifest,
+  entry,
+  nonce,
+  preloads,
+  scripts,
+}: {
+  manifest: Manifest;
+  entry: string;
+  nonce?: string;
+  preloads: Map<string, Preload>;
+  scripts: Map<string, InjectableScript>;
+}) => {
+  if (!manifest) {
+    return;
+  }
+  const entryChunk = manifest[entry];
+  if (!entryChunk) {
+    console.error('Entry chunk not found in manifest', entry);
+    return;
+  }
+
+  if (entryChunk.css) {
+    for (const chunk of entryChunk.css) {
+      addStylesheetToPreloads({ preloads, chunk, nonce });
+    }
+  }
+  if (entryChunk.assets) {
+    for (const chunk of entryChunk.assets) {
+      addAssetToPreloads({ preloads, chunk, nonce });
+    }
+  }
+
+  addFileToPreloads({ preloads, entry, entryChunk, nonce });
+
+  scripts.set(entry, {
+    src: entryChunk.file,
+    isEntry: Boolean(entryChunk.isEntry),
+    nonce,
+  });
+
+  if (entryChunk.imports) {
+    for (const chunk of entryChunk.imports) {
+      if (!preloads.has(chunk)) {
+        parseManifestForEntry({
+          manifest,
+          entry: chunk,
+          nonce,
+          preloads,
+          scripts,
+        });
+      }
+    }
+  }
+};
+
+const addFileToPreloads = ({
+  preloads,
+  entryChunk,
+  entry,
+  nonce,
+}: {
+  preloads: Map<string, Preload>;
+  entryChunk: any;
+  entry: string;
+  nonce?: string;
+}) => {
+  preloads.set(entry, {
+    rel: 'modulepreload',
+    href: entryChunk.file,
+    nonce,
+  });
+};
+
+const addStylesheetToPreloads = ({
+  preloads,
+  chunk,
+  nonce,
+}: {
+  preloads: Map<string, Preload>;
+  chunk: string;
+  nonce?: string;
+}) => {
+  preloads.set(chunk, {
+    rel: 'stylesheet',
+    href: chunk,
+    type: 'text/css', // important for link.
+    nonce,
+  });
+};
+
+const addAssetToPreloads = ({
+  preloads,
+  chunk,
+  nonce,
+}: {
+  preloads: Map<string, Preload>;
+  chunk: string;
+  nonce?: string;
+}) => {
+  const ext = extname(chunk).substring(1);
+  let as;
+  let mimeType;
+
+  switch (ext) {
+    case 'png':
```

Comment:
This will need to support any image formats sku supports. Looks like `bmp` and `avif` are missing, and worth adding `jpeg` too.